include/ignition/gazebo/comms/Broker.hh

```diff
@@ -0,0 +1,156 @@
+  /// to be delivered to one of the destination, it'll be stored in the inbound
+  /// queue of the destination's address.
+  ///
+  /// The main goal of this class is to receive the comms requests, stamp the
+  /// time, and place them in the appropriate outbound queue, as well as deliver
+  /// the messages that are in the inbound queues.
+  ///
+  /// The instance of the comms model is responsible for moving around the
+  /// messages from the outbound queues to the inbound queues.
+  ///
+  /// The broker can be configured with the following SDF parameters:
+  ///
+  /// * Optional parameters:
+  /// <broker> Element used to capture the broker parameters. This block can
+  ///          contain any of the next parameters:
+  ///    <messages_topic>: Topic name where the broker receives all the incoming
+  ///                      messages. The default value is "/broker/msgs"
+  ///    <bind_service>: Service name used to bind an address.
+  ///                    The default value is "/broker/bind"
+  ///    <unbind_service>: Service name used to unbind from an address.
+  ///                      The default value is "/broker/unbind"
+  ///
+  /// Here's an example:
+  /// <plugin
+  ///   filename="ignition-gazebo-perfect-comms-system"
+  ///   name="ignition::gazebo::systems::PerfectComms">
+  ///   <broker>
+  ///     <messages_topic>/broker/inbound</messages_topic>
+  ///     <bind_service>/broker/bind_address</bind_service>
+  ///     <unbind_service>/broker/unbind_address</unbind_service>
+  ///   </broker>
+  /// </plugin>
+  class Broker
+  {
+    /// \brief Constructor.
+    public: Broker();
+
+    /// \brief Destructor.
+    public: virtual ~Broker();
+
+    /// \brief Configure the broker via SDF.
+    /// \param[in] _sdf The SDF Element associated with the broker parameters.
+    public: void Load(std::shared_ptr<const sdf::Element> _sdf);
+
+    /// \brief Start handling comms services.
+    ///
+    /// This function allows us to wait to advertise capabilities to
+    /// clients until the broker has been entirely initialized.
+    public: void Start();
+
+    /// \brief Get the current time.
+    /// \return Current time.
+    public: std::chrono::steady_clock::duration Time() const;
+
+    /// \brief Set the current time.
+    /// \param[in] _time Current time.
+    public: void SetTime(const std::chrono::steady_clock::duration &_time);
+
+    /// \brief This method associates an address with a client topic used as
+    /// callback for receiving messages. This is a client requirement to
+    /// start receiving messages.
+    /// \param[in] _req Bind request containing the following content:
+    ///   _req[0] Client address.
+    ///   _req[1] Model name associated to the address.
+    ///   _req[2] Client subscription topic.
+    /// \param[out] _rep Unused
+    /// \return True when the bind service succeeded or false otherwise.
+    public: bool OnBind(const ignition::msgs::StringMsg_V &_req,
```

Comment:
Is StringMsg_V really a good choice here?

Reply:
I totally agree but until we have a concrete alternative we'd need to go with the current approach.